tools/Jenkins/dependency_spotter.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import re
+import urllib.request
+
+
+def dependency_spotter(os_name: str, first_build: int, second_build: int, pipeline: str, log_file: str):
+    if second_build < first_build:
+        yes_no = input("It looks like the second build is older than the first build, do you want to swap the order? (y/n)")
+        if yes_no.lower() == "y":
+            first_build, second_build = second_build, first_build
+
+    if log_file:
+        compare_dependencies_for_file(os_name, first_build, second_build, pipeline, log_file)
+        return
+
+    first_build_log_files = extract_available_log_files(os_name, first_build, pipeline)
+    second_build_log_files = extract_available_log_files(os_name, second_build, pipeline)
+
+    files_in_both_builds = []
+    files_missing_from_second_build = []
+    for file in first_build_log_files:
+        list_to_append = files_in_both_builds if file in second_build_log_files else files_missing_from_second_build
+        list_to_append.append(file)
+
+    print("Files available for comparison:")
+    if len(files_in_both_builds) > 0:
+        for i, file in enumerate(files_in_both_builds):
+            print(f"{i+1}) {file}")
+    else:
+        print("None")
+    print("")
+    print("Files missing from second build:")
+    if len(files_missing_from_second_build) > 0:
+        for file in files_missing_from_second_build:
+            print(file)
+    else:
+        print("None")
+    print("")
+
+    if len(files_in_both_builds) == 0:
+        return
+
+    files_to_compare = input(
+        "Which files do you want to compare? (Either a single number, a comma-separated list of numbers, or 0 for all):"
+    )
+    if files_to_compare == "":
+        return
```

Comment:
Might be nicer if leaving this option blank (just pressing return) compared all files instead of exiting. I don't think returning should be an option here unless it's stated in the message to the user.

Also, I know I suggested this idea, but possibly it's an over complication and the user should just specify the log file(s) they want to compare in the parsed args (default can be all).